File: backend/src/main/java/org/learn/todo/service/TodoService.java
package org.learn.todo.service;

import org.learn.todo.dto.TodoDto;

import java.util.List;

public interface TodoService {
    TodoDto addTodo(TodoDto todoDto);
    TodoDto getTodoById(Long todoId);
    List<TodoDto> getAllTodos();
    TodoDto updateTodo(Long todoId, TodoDto updatedTodoDto);
    void deleteTodo(Long todoId);
    TodoDto completeTodo(Long todoId);
    TodoDto inCompleteTodo(Long todoId);
}

File: frontend/src/service/TodoService.js
import axios from 'axios'
import { getToken } from './AuthService'

const REST_API_BASE_URL = "http://localhost:8080/api/todos"

axios.interceptors.request.use(function (config) {

    config.headers['Authorization'] = getToken();

    return config
}), function (error) {
    return Promise.reject(error);
}

export const getAllTodos = () => axios.get(REST_API_BASE_URL);

export const addTodo = (todo) => axios.post(REST_API_BASE_URL, todo);

export const getTodo = (todoId) => axios.get(REST_API_BASE_URL + "/" + todoId);

export const updateTodo = (todoId, todo) => axios.put(REST_API_BASE_URL + "/" + todoId, todo);

export const deleteTodo = (todoId) => axios.delete(REST_API_BASE_URL + "/" + todoId);

export const completeTodo = (todoId) => axios.patch(REST_API_BASE_URL + "/" + todoId + "/complete");

export const inCompleteTodo = (todoId) => axios.patch(REST_API_BASE_URL + "/" + todoId + "/in-complete");